src/box/ck_constraint.c

```diff
@@ -1,282 +0,0 @@
- *
- * 1. Redistributions of source code must retain the above
- *    copyright notice, this list of conditions and the
- *    following disclaimer.
- *
- * 2. Redistributions in binary form must reproduce the above
- *    copyright notice, this list of conditions and the following
- *    disclaimer in the documentation and/or other materials
- *    provided with the distribution.
- *
- * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
- * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
- * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
- * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
- * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
- * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
- * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
- * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
- * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
- * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
- * SUCH DAMAGE.
- */
-#include "box/session.h"
-#include "execute.h"
-#include "bind.h"
-#include "ck_constraint.h"
-#include "errcode.h"
-#include "schema.h"
-#include "small/region.h"
-#include "sql.h"
-#include "sql/sqlInt.h"
-#include "sql/vdbeInt.h"
-#include "tuple.h"
-
-const char *ck_constraint_language_strs[] = {"SQL"};
-
-struct ck_constraint_def *
-ck_constraint_def_new(const char *name, uint32_t name_len, const char *expr_str,
-		      uint32_t expr_str_len, uint32_t space_id,
-		      enum ck_constraint_language language, bool is_enabled)
-{
-	uint32_t expr_str_offset;
-	uint32_t ck_def_sz = ck_constraint_def_sizeof(name_len, expr_str_len,
-						      &expr_str_offset);
-	struct ck_constraint_def *ck_def =
-		(struct ck_constraint_def *) malloc(ck_def_sz);
-	if (ck_def == NULL) {
-		diag_set(OutOfMemory, ck_def_sz, "malloc", "ck_def");
-		return NULL;
-	}
-	ck_def->is_enabled = is_enabled;
-	ck_def->expr_str = (char *)ck_def + expr_str_offset;
-	ck_def->language = language;
-	ck_def->space_id = space_id;
-	memcpy(ck_def->expr_str, expr_str, expr_str_len);
-	ck_def->expr_str[expr_str_len] = '\0';
-	memcpy(ck_def->name, name, name_len);
-	ck_def->name[name_len] = '\0';
-	return ck_def;
-}
-
-void
-ck_constraint_def_delete(struct ck_constraint_def *ck_def)
-{
-	free(ck_def);
-}
-
-/**
- * Resolve space_def references for check constraint via AST
- * tree traversal.
- * @param expr Check constraint AST object to update.
- * @param space_def Space definition to use.
- * @retval 0 On success.
- * @retval -1 On error.
- */
-static int
-ck_constraint_resolve_field_names(struct Expr *expr,
-				  struct space_def *space_def)
-{
-	struct Parse parser;
-	sql_parser_create(&parser, sql_get(), default_flags);
-	parser.parse_only = true;
-	sql_resolve_self_reference(&parser, space_def, NC_IsCheck, expr);
```

Comment:
AFAIS, `NC_IsCheck` is never set. Can probably drop it and all its checks.

Reply:
Thanks! Dropped.